### PR TITLE
Remove tags from search

### PR DESCRIPTION
**Pull request by Ghalko**
## Proposed
Adds x's to the upper right corners of tags. These can be clicked which adds ```NOT tags:("<tag>")``` to the search.

These should combine into ANDs and NOTs but otherwise this is mostly done. Partially addresses #1266.

## Problems
- Should the x's be hidden when cursor is not over?
- x background has part of the tag in it, this could be "solved" by the previous thing.

### CODE REVIEW

File: website/static/js/search.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ var ViewModel = function(params) {
         }
     };
 
-    self.addTag = function(name) {
+    self.addTag = function(name, action) {
```
Comment (sloria) [inline website/static/js/search.js]: Having an `addTag` where `action` can be `remove` is untuitive. Why not a `removeTag` method?

Comment (Ghalko) [inline website/static/js/search.js]: Changed name to clickTag, kept actions.

**Comment by Ghalko** [comment]
Simply offers an x to click on to remove a tag from a search.

**Comment by samanehsan** [comment]
@Ghalko, can you bring this up to date with develop? Thanks! :palm_tree:

**Comment by Ghalko** [comment]
@caneruguz suggestions? The x's turn red when hover. (Probably the wrong red.)

### Current
------
screen shot 2015-06-10 at 11 49 43 am
### New
-----
screen shot 2015-06-10 at 11 43 54 am

**Comment by sloria** [comment]
ping @caneruguz

**Comment by caneruguz** [comment]
My suggestions are:
- They should not be permanent, but visible on hover (just make sure the hover covers them and users can actually click them.
- Color should not be red but a muted grey.

**Comment by sloria** [comment]
@Ghalko When you get the chance, can you please implement Caner's suggestions?

**Comment by Ghalko** [comment]
@sloria Sorry about the merge, it was really far behind. Changed to incorporate @caneruguz suggestions.

**Pull request merged**